tests/BenchmarkDotNet.IntegrationTests/InProcessTest.cs

```diff
@@ -0,0 +1,388 @@
+                .With(Job.Dry.With(new InProcessToolchain(TimeSpan.Zero, codegenMode, true)).WithInvocationCount(UnrollFactor).WithUnrollFactor(UnrollFactor))
+                .With(logger ?? (Output != null ? new OutputLogger(Output) : ConsoleLogger.Default))
+                .With(DefaultColumnProviders.Instance);
+        }
+
+        [Fact]
+        public void InProcessBenchmarkAllCasesReflectionEmitTest()
+        {
+            var logger = new OutputLogger(Output);
+            var config = CreateInProcessConfig(BenchmarkActionCodegen.ReflectionEmit, logger);
+
+            try
+            {
+                BenchmarkAllCases.Counter = 0;
+
+                var summary = CanExecute<BenchmarkAllCases>(config);
+
+                var testLog = logger.GetLog();
+                Assert.Contains("// Benchmark: BenchmarkAllCases.InvokeOnceVoid:", testLog);
+                Assert.DoesNotContain("No benchmarks found", logger.GetLog());
+
+                // Operations + Setup + Cleanup
+                var expectedCount = summary.Reports.SelectMany(r => r.AllMeasurements).Sum(m => m.Operations + 2);
+                Assert.Equal(expectedCount, BenchmarkAllCases.Counter);
+            }
+            finally
+            {
+                BenchmarkAllCases.Counter = 0;
+            }
+        }
+
+        [Fact]
+        public void InProcessBenchmarkAllCasesDelegateCombineTest()
+        {
+            var logger = new OutputLogger(Output);
+            var config = CreateInProcessConfig(BenchmarkActionCodegen.DelegateCombine, logger);
+
+            try
+            {
+                BenchmarkAllCases.Counter = 0;
+
+                var summary = CanExecute<BenchmarkAllCases>(config);
+
+                var testLog = logger.GetLog();
+                Assert.Contains("// Benchmark: BenchmarkAllCases.InvokeOnceVoid:", testLog);
+                Assert.DoesNotContain("No benchmarks found", logger.GetLog());
+
+                // Operations + Setup + Cleanup
+                var expectedCount = summary.Reports.SelectMany(r => r.AllMeasurements).Sum(m => m.Operations + 2);
+                Assert.Equal(expectedCount, BenchmarkAllCases.Counter);
+            }
+            finally
+            {
+                BenchmarkAllCases.Counter = 0;
+            }
+        }
+
+        [Fact]
+        public void InProcessBenchmarkAllCasesDiagnoserTest()
+        {
+
+            var logger = new OutputLogger(Output);
+            var config = new ManualConfig()
+                .With(Job.Default.With(InProcessToolchain.Instance))
+                .With(MemoryDiagnoser.Default)
+                .With(logger)
+                .With(DefaultColumnProviders.Instance);
+            try
+            {
+                BenchmarkAllCases.Counter = 0;
+
+                var summary = CanExecute<BenchmarkAllocates>(config);
+
+                var testLog = logger.GetLog();
+                Assert.Contains("// Benchmark: BenchmarkAllocates.Allocates:", testLog);
```

Comment:
it would be better if you could use the strong-typed properties of the summary instead of looking for strings in the output